src/catalog/hooks/useCourseData.ts

```diff
@@ -0,0 +1,68 @@
+import { useState, useCallback, useEffect } from 'react';
+
+import type { CourseListSearchResponse } from '@src/data/course-list-search/types';
+import type { UseCourseDataProps } from './types';
+
+/**
+ * Custom hook for managing course data caching and search result handling.
+ *
+ * This hook provides functionality to:
+ * - Cache previous course data when not searching
+ * - Handle search result states (successful results vs no results)
+ * - Manage data persistence for better UX
+ * - Coordinate with search state management
+ */
+export const useCourseData = ({
+  courseData,
+  searchString,
+  isFetching,
+  onNoSearchResults,
+  onClearLastSearchQuery,
+}: UseCourseDataProps) => {
+  const [previousCourseData, setPreviousCourseData] = useState<CourseListSearchResponse | null>(null);
+
+  /**
+   * Saves course data to cache when not actively searching.
+   */
+  const savePreviousCourseData = useCallback((data: CourseListSearchResponse) => {
+    if (data && !searchString) {
+      setPreviousCourseData(data);
```

Comment:
[inform]: Because we are implementing pagination, we do not store the data for all courses available on the platform in the component state.